docker-mixin/lib/custom-barchart-grafonnet/custom-barchart.libsonnet

```diff
@@ -0,0 +1,162 @@
+local grafana = import 'github.com/grafana/grafonnet-lib/grafonnet/grafana.libsonnet';
+
+{
+  new(q1, q2, q3)::
+    {
+      datasource: {
+        type: 'loki',
+        uid: '${loki_datasource}',
+      },
+      fieldConfig: {
+        defaults: {
+          color: {
+            mode: 'fixed',
+          },
+          custom: {
+            axisLabel: '',
+            axisPlacement: 'auto',
+            axisSoftMin: 0,
+            fillOpacity: 50,
+            gradientMode: 'none',
+            hideFrom: {
+              legend: false,
+              tooltip: false,
+              viz: false,
+            },
+            lineWidth: 1,
+            scaleDistribution: {
+              type: 'linear',
+            },
+          },
+          mappings: [],
+          thresholds: {
+            mode: 'absolute',
+            steps: [
+              {
+                color: 'green',
+                value: null,
+              },
+            ],
+          },
+          unit: 'short',
+        },
+        overrides: [
+          {
+            matcher: {
+              id: 'byFrameRefID',
+              options: 'A',
+            },
+            properties: [
+              {
+                id: 'displayName',
+                value: 'Lines',
+              },
+              {
+                id: 'color',
+                value: {
+                  fixedColor: 'super-light-blue',
+                  mode: 'fixed',
+                },
+              },
+            ],
+          },
+          {
+            matcher: {
+              id: 'byFrameRefID',
+              options: 'B',
+            },
+            properties: [
+              {
+                id: 'displayName',
+                value: 'Warnings',
+              },
+              {
+                id: 'color',
+                value: {
+                  fixedColor: 'orange',
+                  mode: 'fixed',
+                },
+              },
+            ],
+          },
+          {
+            matcher: {
+              id: 'byFrameRefID',
+              options: 'C',
+            },
+            properties: [
+              {
+                id: 'displayName',
+                value: 'Errors',
+              },
+              {
+                id: 'color',
+                value: {
+                  fixedColor: 'red',
+                  mode: 'fixed',
+                },
+              },
+            ],
+          },
+        ],
+      },
+      gridPos: {
+        h: 0,
+        w: 0,
+        x: 0,
+        y: 0,
+      },
+      id: 12,
```

Comment:
If this gets used in another dashboard, or even if it's order in the existing dashboard gets changed, the ID could end up conflicting.

Best to exclude this as well, and count on the `grafana.dashboard.addPanel` to add the ID.

Reply:
Thanks for the catch! Removed the id.